### PR TITLE
list Environments based on the promotion

### PR DESCRIPTION
**Pull request by sahil143**
## Fixes 

https://issues.redhat.com/browse/HAC-1967

## Description

Sort Environments in promotion order

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/Environment/ApplicationEnvironmentCards.tsx [hunk 1]
```diff
@@ -28,7 +28,11 @@ const ApplicationEnvironmentCard: React.FC<ApplicationEnvironmentCardProps> = ({
 }) => {
   const strategy = getEnvironmentDeploymentStrategyLabel(environment);
   return (
-    <Card isCompact isExpanded={isExpanded}>
+    <Card
+      isCompact
+      isExpanded={isExpanded}
+      style={{ marginBottom: 'var(--pf-global--spacer--sm)' }}
```
Comment (rottencandy) [inline src/components/Environment/ApplicationEnvironmentCards.tsx:34]: Nit: Can use token here
`import { global_spacer_sm } from '@patternfly/react-tokens/dist/js/global_spacer_sm';`

**Comment by sahil143** [comment]
image

**Comment by christianvogt** [comment]
Should we not also sort alphabetically after sorting by promotion order because multiple envs can have the same parent?

**Comment by sahil143** [comment]
image

**Review by rottencandy — APPROVED** [review]
/lgtm

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Pull request merged**